src/os/ObjectStore.h

```diff
@@ -504,10 +511,21 @@ class ObjectStore {
     void set_fadvise_flags(uint32_t flags) {
       data.fadvise_flags = flags;
     }
+
     void set_fadvise_flag(uint32_t flag) {
-      data.fadvise_flags = data.fadvise_flags | flag;
+      if (use_own_fadvise_flags) {
+	int ops = data.ops;
+	char *buffer = op_bl.get_contiguous(0, ops * sizeof(Op));
+	for (int i = 0; i < ops; i++) {
+	  Op *_op = (Op *)buffer;
+	  if (_op->op == OP_WRITE)
+	    _op->fadvise_flags |= flag;
+	  buffer += sizeof(Op);
+	}
```

Comment:
this seems awkward.  we should perhaps look at the callers and see if they shoudl be more specific.  today WRITE is the only op to take the flags but i think almost every op would benefit here... reads, certainly.  also omap get/set could pass those hints down to a smarter backend...